src/ameba/rule/naming/ascii_identifiers.cr

```diff
@@ -0,0 +1,68 @@
+module Ameba::Rule::Naming
+  # A rule that reports non-ascii characters in identifiers.
+  #
+  # Favour this:
+  #
+  # ```
+  # class BigAwesomeWolf
+  # end
+  # ```
+  #
+  # Over this:
+  #
+  # ```
+  # class BigAwesome🐺
+  # end
+  # ```
+  #
+  # YAML configuration example:
+  #
+  # ```
+  # Naming/AsciiIdentifiers:
+  #   Enabled: true
+  # ```
+  class AsciiIdentifiers < Base
+    include AST::Util
+
+    properties do
+      description "Disallows non-ascii characters in identifiers"
+    end
+
+    MSG = "Identifier contains non-ascii characters"
+
+    def test(source, node : Crystal::Assign)
+      if (target = node.target).is_a?(Crystal::Path)
+        check_issue(source, target, target.to_s)
+      end
+    end
+
+    def test(source, node : Crystal::Def)
+      location = name_location(node)
+      end_location = name_end_location(node)
+
+      if location && end_location
+        check_issue(source, location, end_location, node.name)
+      else
+        check_issue(source, node, node.name)
+      end
+    end
+
+    def test(source, node)
+      if node.responds_to?(:name) && (name = node.name)
```

Comment:
not sure if that is a good idea to visit all the nodes and invoke responds_to? on them. Can this slow down process?

Reply:
The meat of `responds_to` is a compile-time operation. At runtime it's just a type filter (like `if x.is_a?(Foo)`).